tests/Database/DatabaseEloquentModelTest.php

```diff
@@ -2214,8 +2214,10 @@ public function testIntKeyTypePreserved()
     public function testStringKeyTypePreserved()
     {
         $model = $this->getMockBuilder(EloquentKeyTypeModelStub::class)->onlyMethods(['newModelQuery', 'updateTimestamps', 'refresh'])->getMock();
+        $model->id = 'string id';
+
         $query = m::mock(Builder::class);
-        $query->shouldReceive('insertGetId')->once()->with([], 'id')->andReturn('string id');
+        $query->shouldReceive('insert')->once()->with(['id' => 'string id']);
         $query->shouldReceive('getConnection')->once();
         $model->expects($this->once())->method('newModelQuery')->willReturn($query);
```

Comment:
I had to rewrite the test here because of the change I did in the `getIncrementing` method. I think the current test wasn't realistic as I can't see how you can increment with string-based keys. With string-based keys you always need to set them beforehand afaik (like I adjusted the test to).